server/src/test/java/org/elasticsearch/common/geo/CentroidCalculatorTests.java

```diff
@@ -44,12 +48,30 @@ public void test() {
             System.arraycopy(y, 0, subY, 0, i + 1);
             Geometry geometry  = new Line(subX, subY);
             calculator = new CentroidCalculator(geometry);
-            assertThat(calculator.getX(), equalTo(xRunningAvg[i]));
-            assertThat(calculator.getY(), equalTo(yRunningAvg[i]));
+            assertEquals(xRunningAvg[i], calculator.getX(), 0.000001);
+            assertEquals(yRunningAvg[i], calculator.getY(), 0.000001);
         }
         CentroidCalculator otherCalculator = new CentroidCalculator(new Point(0, 0));
         calculator.addFrom(otherCalculator);
-        assertThat(calculator.getX(), equalTo(50.0));
-        assertThat(calculator.getY(), equalTo(5.0));
+        assertEquals(55.0, calculator.getX(), 0.0000001);
+        assertEquals(5.5, calculator.getY(), 0.0000001);
+    }
+
+    public void testPolyonWithHole() throws Exception {
+        Polygon polyWithHole = new Polygon(new LinearRing(new double[]{-50, 50, 50, -50, -50}, new double[]{-50, -50, 50, 50, -50}),
+            Collections.singletonList(new LinearRing(new double[]{-30, -30, 30, 30, -30}, new double[]{-30, 30, 30, -30, -30})));
+        CentroidCalculator calculator = new CentroidCalculator(polyWithHole);
+        assertThat(calculator.getX(), equalTo(0.0));
+        assertThat(calculator.getY(), equalTo(0.0));
+        assertThat(calculator.sumWeight(), equalTo(6400.0));
+    }
+
+    public void testPolygonWithEqualSizedHole() {
```

Comment:
This test is ok but I think it cannot / should not happen, it is an invalid polygon.

Reply:
makes sense, since the calculator accepts it, I wanted to be sure of the behavior. do you think it can be removed?

Reply:
leave it, it makes sense as it checks logic is correct